### PR TITLE
feat: Handle zero max cap on chain logic

### PR DESCRIPTION
**Pull request by trinitys7**
Following with https://github.com/osmosis-labs/mesh-security/pull/192, a new logic for storing delegator data is required in x/meshsecurity module.

### CODE REVIEW

File: docs/proto/proto-docs.md [hunk 1]
```diff
@@ -238,6 +257,21 @@ Query provides defines the gRPC querier service
...
 
+<a name="osmosis.meshsecurity.v1beta1.ScheduledWork"></a>
+
+### ScheduledWork
+
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| `repeat` | [bool](#bool) |  |  |
```
Comment (JakeHartnell) [inline docs/proto/proto-docs.md:268]: Description for this field / section would be nice. What is `ScheduledWork`? What is this `repeat` field?

File: x/meshsecurity/keeper/keeper.go [hunk 2]
```diff
@@ -147,6 +147,66 @@ func (k Keeper) setTotalDelegated(ctx sdk.Context, actor sdk.AccAddress, newAmou
...
+// setDelegation store the delegation of a given delegator bond with validator
+func (k Keeper) setDelegation(ctx sdk.Context, actor, delAddr sdk.AccAddress, valAddr sdk.ValAddress, changeAmount math.Int) {
+	store := ctx.KVStore(k.storeKey)
+
+	newDelegation := k.GetDelegation(ctx, k.storeKey, actor, delAddr, valAddr)
+	newDelegation.Amount = newDelegation.Amount.Add(changeAmount)
+	if !newDelegation.Amount.IsPositive() {
+		store.Delete(types.BuildDelegationKey(actor, delAddr, valAddr))
+		return
+	}
```
Comment (JakeHartnell) [inline x/meshsecurity/keeper/keeper.go:201]: A little confused here, though I'm a Rust dev not a Go dev, we remove a key from the store if it's negative here? Why would it ever be negative?

Assume `IsPositive()` is false for zero?

**Comment by JakeHartnell** [comment]
This PR looks good to me, though the title is confusing. Where do you "Handle zero max cap on chain logic"?

**Comment by JakeHartnell** [comment]
A few minor comments, otherwise LGTM. Would love for a better Cosmos SDK dev to have a look though too.

**Review by vuong177 — APPROVED** [review]

**Pull request merged**